packages/medusa/src/strategies/product-export.ts

```diff
@@ -0,0 +1,781 @@
+            moneyAmount?.currency_code?.toLowerCase() ?? "",
+          entityName: "moneyAmount",
+          index: i,
+        })
+        .set(`Price ${i + 1} Region name`, {
+          accessor: (moneyAmount: MoneyAmount) =>
+            moneyAmount?.region?.name?.toLowerCase() ?? "",
+          entityName: "moneyAmount",
+          index: i,
+        })
+        .set(`Price ${i + 1} Amount`, {
+          accessor: (moneyAmount: MoneyAmount) =>
+            moneyAmount?.amount?.toString() ?? "",
+          entityName: "moneyAmount",
+          index: i,
+        })
+    }
+  }
+
+  private buildProductLines(product: Product): string[] {
+    const outputLineData: string[] = []
+
+    const productLine = this.buildProductLineData(product)
+    outputLineData.push(productLine)
+
+    const variantLines = this.buildProductVariantsLineData(
+      product,
+      product.variants
+    )
+    outputLineData.push(...variantLines)
+
+    return outputLineData
+  }
+
+  private buildProductLineData(product: Product): string {
+    const productLineData: string[] = []
+    for (const [, columnSchema] of this.columnDescriptors.entries()) {
+      if (columnSchema.entityName === "product") {
+        productLineData.push(columnSchema.accessor(product))
+        continue
+      }
+      if (
+        columnSchema.entityName === "option" &&
+        columnSchema.index !== undefined
+      ) {
+        const option = product?.options[columnSchema.index] ?? {}
+        productLineData.push(columnSchema.accessor(option))
+        continue
+      }
+      if (
+        columnSchema.entityName === "image" &&
+        columnSchema.index !== undefined
+      ) {
+        const image = product?.images[columnSchema.index] ?? {}
+        productLineData.push(columnSchema.accessor(image))
+        continue
+      }
+      productLineData.push("")
+    }
+
+    productLineData.push(this.NEWLINE)
+    return productLineData.join(this.DELIMITER_)
+  }
+
+  private buildProductVariantsLineData(
+    product: Product,
+    variants: ProductVariant[] = []
+  ): string[] {
+    const outputLineData: string[] = []
+
+    for (const variant of variants) {
+      const variantLineData: string[] = []
+
+      for (const [
+        columnTitle,
+        columnSchema,
+      ] of this.columnDescriptors.entries()) {
+        if (
+          columnTitle === "Product Handle" &&
+          columnSchema.entityName === "product"
+        ) {
+          variantLineData.push(columnSchema.accessor(product))
+          continue
+        }
+
+        if (columnSchema.entityName === "variant") {
+          variantLineData.push(columnSchema.accessor(variant))
+          continue
+        }
+        if (
+          columnSchema.entityName === "variantOption" &&
+          columnSchema.index !== undefined
+        ) {
+          const option = variant?.options[columnSchema.index] ?? {}
+          variantLineData.push(columnSchema.accessor(option))
+          continue
+        }
```

Comment:
same as above

Reply:
True, i will do like i was doing before ^^ which was exactly what you are saying 😅